apps/web/src/pages/templates/editor_v2/CloudWorkflowSettingsSidePanel.tsx

```diff
@@ -16,10 +16,10 @@ export const CloudWorkflowSettingsSidePanel: FC<CloudWorkflowSettingsSidePanelPr
   const { templateId: workflowId = '' } = useParams<{ templateId: string }>();
   const [searchParams] = useSearchParams();
   const { isLoading, workflowUserPreferences, workflowResourcePreferences } = useCloudWorkflowPreferences(workflowId);
-  const { setValue } = useFormContext<WorkflowDetailFormContext>();
+  const { setValue, getValues } = useFormContext<WorkflowDetailFormContext>();
 
   useEffect(() => {
-    if (workflowUserPreferences !== undefined) {
+    if (workflowUserPreferences !== undefined && getValues('preferences') === undefined) {
```

Comment:
Using the form state to retain context of the form values, only setting the form values if it wasn't set already. This prevents the Preferences component state from being reset to `null` when the API fetch occurs on component render.